stl/inc/iterator

```diff
@@ -133,17 +133,23 @@ public:
     using traits_type  = _Traits;
     using istream_type = basic_istream<_Elem, _Traits>;
 
-    constexpr istream_iterator() : _Myistr(nullptr), _Myval() {}
+    static_assert(conjunction_v<is_default_constructible<_Ty>, is_copy_constructible<_Ty>, is_copy_assignable<_Ty>>,
+        "The type T shall meet the Cpp17DefaultConstructible, Cpp17CopyConstructible, and Cpp17CopyAssignable "
+        "requirements. (Enabled unconditionally from CXX20)");

```

Comment:
* While this is directly quoting the Standardese, we've avoided mentioning the Cpp17-prefixed requirements to users (even the Standard will be editorially changed to avoid these verbose names soon). Just saying "default constructible, copy constructible, and copy assignable" would be better.
* It would be better to begin with `"istream_iterator<T> requires T to be"` so the user has an idea of what the mentioned `T` is. (The template instantiation context, which many users don't look at, provides such information, but it will mention several template parameters, none of which are named literally `T`.)
* The "Enabled unconditionally from CXX20" explanation should be changed to a Standardese citation; here it's `N4835 [istream.iterator]/2` (We don't really need to explain that we're enforcing this unconditionally; also, when we need to mention Standard versions to users, we say C++20 instead of CXX20.)

For comparison, see:
https://github.com/microsoft/STL/blob/d0d724fb916c59f1c4f2830330ed967a72d2d2b8/stl/inc/variant#L1133-L1134

Reply:
That is definitely a great improvement